### PR TITLE
drivers: usb_dc_dw: cleanup, move USB_DW macros to driver code, use on additional platforms

### PR DESCRIPTION
**Pull request by jfischer-no**
The goal of the PR is to cleanup and improve usb_dc_dw driver, make it usable on additional platforms, and prepare it for porting to a new experimental UDC API.

### CODE REVIEW

File: drivers/usb/device/usb_dw_registers.h
Comment (tmon-nordic) [inline drivers/usb/device/usb_dw_registers.h:160]:
```suggestion
#define USB_DW_GHWCFG2_HSPHYTYPE_SHIFT		6
```

File: drivers/usb/device/usb_dw_registers.h
Comment (tmon-nordic) [inline drivers/usb/device/usb_dw_registers.h:145]: The calculation is incorrect, e.g. EP 15 direction is on bits [31:30], but `(0x3 << (15 << 2))` is way outside 32-bit integer.
```suggestion
#define USB_DW_GHWCFG1_EPDIR_MASK(i)		(0x3 << (i * 2))
#define USB_DW_GHWCFG1_EPDIR_SHIFT(i)		(i * 2)
```

While SHIFT makes more sense, the convention in this header seems to be OFFSET instead of SHIFT.

File: drivers/usb/device/usb_dw_registers.h
Comment (tmon-nordic) [inline drivers/usb/device/usb_dw_registers.h:168]:
```suggestion
#define USB_DW_GHWCFG3_DFIFODEPTH_MASK	(0xFFFF << 16)
#define USB_DW_GHWCFG3_DFIFODEPTH_SHIFT	16
```

File: drivers/usb/device/usb_dw_registers.h
Comment (tmon-nordic) [inline drivers/usb/device/usb_dw_registers.h:172]:
```suggestion
#define USB_DW_GHWCFG4_INEPS_MASK		(0xF << 26)
#define USB_DW_GHWCFG4_INEPS_SHIFT		26
```

**Review by tmon-nordic — CHANGES_REQUESTED** [review]
Perhaps the cleanup should also include renaming `OFFSET` suffix to `SHIFT`. Currently the new definitions use the `SHIFT` suffix, but old ones use `OFFSET`.

**Review by mbolivar-nordic — COMMENTED** [review]
DT portions seem reasonable to me. I will leave it to the USB maintainers to decide what to do about the old compatible (it looks like the driver will no longer be auto-enabled when the outdated, zephyr-specific compatible is used, right?)

File: drivers/usb/device/usb_dc_dw.c
Comment (anangl) [inline drivers/usb/device/usb_dc_dw.c:102]:
```suggestion
	const struct pinctrl_dev_config *const pcfg = config->pcfg;
```

File: drivers/usb/device/usb_dc_dw.c
Comment (anangl) [inline drivers/usb/device/usb_dc_dw.c:144]:
```suggestion
		    ((void *)PINCTRL_DT_INST_DEV_CONFIG_GET(n)), (NULL))
```

File: drivers/usb/device/usb_dc_dw.c
Comment (anangl) [inline drivers/usb/device/usb_dc_dw.c:66]: Why is this not `const struct pinctrl_dev_config *`?

File: drivers/usb/device/usb_dw_registers.h
Comment (jfischer-no) [reply drivers/usb/device/usb_dw_registers.h]: This uses the wrong name and is fixed in the next commit.

**Review by tmon-nordic — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**
